Feat: 거래/제안내역 CRUD 기능 완료 #5

haeminchoi2 opened the pull request:
## PR 타입(하나 이상의 PR 타입을 선택해주세요)
- [ ] 디자인 변경
- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트

## 반영 브랜치
feature/5-specificDeal -> deal

## 변경 사항
1. 거래/제안내역 CRUD 구현
2. 거래/제안내역 생성 기능
    - [x] 로그인한 유저인지 검증
    - [x] 이미 제안한 내역인지 확인
3. 거래/제안내역 조회 기능
    - [x] 전체 서비스에 대한 거래내역 목록 조회
    - [x] 거래/제안내역 상세 조회
    - [x] 특정 조건 거래/제안내역 목록 조회
4. 거래/제안내역 수정 기능
    - [x] 로그인한 유저인지 검증
    - [x] 판매자 혹은 구매자인지 유저 검증
5. 거래/제안내역 삭제 기능
    - [x] 로그인한 유저인지 검증
    - [x] 판매자 혹은 구매자인지 유저 검증

close #5 

## 테스트 결과
변경 사항을 설명하는데 도움이 되는 스크린샷이나 참고 자료를 추가해주세요.
![거래내역_생성](https://user-images.githubusercontent.com/96516988/203873314-bc67543b-5a0c-498f-8729-31321ae67e8d.png)
![거래내역_조회](https://user-images.githubusercontent.com/96516988/203873318-2791f188-10c8-4e27-a6c2-426be8206a34.png)
![거래내역_상세조회](https://user-images.githubusercontent.com/96516988/203873321-ab79f439-70b3-4754-8aed-6259f30f083a.png)
![거래내역_수정](https://user-images.githubusercontent.com/96516988/203873325-c1a87fef-f693-4b61-bed9-558bac072786.png)
![거래내역_삭제](https://user-images.githubusercontent.com/96516988/203873327-e2c32ad3-e057-43bd-8506-f76906b41fdf.png)
![특정조건_거래내역_조회](https://user-images.githubusercontent.com/96516988/203873329-bd7cfe0f-30ae-407d-9823-7203303e7c3f.png)

haeminchoi2 committed:
Feat: 거래내역 Deal 모델 생성 #5

- 거래내역 Deal모델 작성했습니다.

haeminchoi2 committed:
Feat: deal/urls.py 생성 및 작성 #5

- deal/urls.py 생성했습니다.
- deal/urls.py 작성했습니다.

haeminchoi2 committed:
Feat: DealSerializer 생성 #5

- deal/serializers.py 생성했습니다.
- DealSerializer 생성했습니다.

haeminchoi2 committed:
Feat: 거래내역 Serializer 수정 #5

- 필요한 필드(구매자/판매자의 username)를 Response에 전달하기 위해 ReadOnlyField 추가했습니다.

haeminchoi2 committed:
Feat: 어드민페이지에 거래내역 생성 #5

- deal/admin.py에 거래내역에 관한 페이지 추가했습니다.

haeminchoi2 committed:
Feat: 거래내역 생성, 거래내역 전체목록 구현 완료 #5

- url : /deal/ 에 해당하는 API구현 완료했습니다.

haeminchoi2 committed:
Feat: 거래내역 상세 조회, 수정, 삭제 완료 #5

- 거래내역 상세 API를 위해, deal/urls.py 수정
- 거래내역 수정/삭제를 위해 DealUpdateSerializer 추가했습니다.
- 거래내역 상세 API 구현했습니다. deal/views.py

haeminchoi2 committed:
Feat: 거래내역 CRUD 검증 부분 #5

- 거래내역 CRUD에서 검증하는 코드 추가했습니다.
  - 요청한 유저가 어떤 유저인지 검증
  - 요청한 유저가 로그인된 유저인지 검증

haeminchoi2 committed:
Feat: 특정 조건의 거래내역 조회 기능 #5

- 특정 조건의 거래/제안내역을 조회할 수 있습니다.
  - 특정 조건은 URL Querystring으로 전달받습니다.

hyunjooooojung reviewed (approved):
거래내역 CRUD 완료하신것 확인했습니다 수고하셨습니다 ! :)

ssuin52 reviewed (approved):
복잡한 기능 구현하느라 고생하셨습니다!

kvkvd12 reviewed (approved):
수고하셨습니다!!!

haeminchoi2 committed:
Fix: Deal 모델 수정 #5

- 기존 코드는 테스트용으로 User모델을 Django기본 유저모델을 임포트했었음.
  - main 브랜치에 병합할 예정이므로, 커스텀 유저모델을 임포트 했습니다.

LeeHyunji committed:
Merge branch 'main' into feature/5-specificDeal

LeeHyunji reviewed (approved):
깔끔한 기능입니다
